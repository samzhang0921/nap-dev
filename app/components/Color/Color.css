.clear {
  clear: both;
}

.colorName {
  composes: designer from '../Designer/Designer.css';
}

.colorCheckbox {
  composes: filterCheckbox from '../Designer/Designer.css';

}
.colorCheckedbox {
  composes: filterCheckedbox from '../Designer/Designer.css';

}
