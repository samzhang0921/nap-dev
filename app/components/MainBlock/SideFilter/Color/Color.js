import React from 'react';
import {render} from 'react-dom';
import style from './Color.css';

export default class Color extends React.Component {
    constructor(props){
      super(props);
      this.state = {
      selectBox:false
    };
      this.getColor = this.getColor.bind(this);
    }

getColor (newColor){
  const { selectBox } = this.state;
   this.setState({
     selectBox: !selectBox
   });
  this.props.updateColorList(newColor)
}
componentWillUpdate(nextProps, nextState){
  if (this.props.clearClicked !== nextProps.clearClicked){
    this.setState({
      selectBox:false
    })
  }
}
render(){
  const checkBox = this.state.selectBox? style.colorCheckedbox: style.colorCheckbox;
  return (
       <li data-category-id={this.props.id} onClick={()=>this.getColor(this.props.id)}>
         <span className={checkBox}></span>
         <span className={style.colorName}>{this.props.name}</span>
         <div className={style.clear}></div>
     </li>
  )
}

}
