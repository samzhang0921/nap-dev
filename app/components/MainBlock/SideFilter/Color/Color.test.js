import React from 'react';
import {mount} from 'enzyme';
import toJson from 'enzyme-to-json';
import renderer from 'react-test-renderer';
import Color from './Color';

describe('Color', () => {
  const id =  1643;
  const name = "blue";

  it('should mount item ', () => {
    const tree = toJson(mount(<Color id={[]}/>));
    expect(tree).toMatchSnapshot();
  });

  it('should match its snapshot with items', () => {

    const tree = renderer.create(<Color id={id} name={name}/>).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should change class on the onClick function', () => {
    const onChange = jest.fn();
    const component = mount(<Color id={id} name={name} updateColorList={onChange}/>);
    component.find('li').simulate('click');
    expect(component.find('.colorCheckedbox')).toHaveLength(1);
  });
});
