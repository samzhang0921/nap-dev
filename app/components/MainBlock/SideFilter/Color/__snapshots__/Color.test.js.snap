// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Color should match its snapshot with items 1`] = `
<li
  data-category-id={1643}
  onClick={[Function]}
>
  <span
    className="colorCheckbox"
  />
  <span
    className="colorName"
  >
    blue
  </span>
  <div
    className="clear"
  />
</li>
`;

exports[`Color should mount item  1`] = `
<Color
  id={Array []}
>
  <li
    data-category-id={Array []}
    onClick={[Function]}
  >
    <span
      className="colorCheckbox"
    />
    <span
      className="colorName"
    />
    <div
      className="clear"
    />
  </li>
</Color>
`;
